### ISSUE TITLE
Grunt build fails on sass.js task on raspberry pi - Unsupported architecture (arm) with Node.js 6.x

### ISSUE THREAD

**Issue by WtfJoke** [AHAAAAAAA/PokemonGo-Map#2918]
If I try to run pokemongomap on my rasppery pi I get an error as soon as I want to build it using the command grunt build.

Heres the log:
'Loading "sass.js" tasks...ERROR
>> Error: Node Sass does not yet support your current environment: Linux Unsupported architecture (arm) with Node.js 6.x
>> For more information on which environments are supported please see:
>> https://github.com/sass/node-sass/releases/tag/v3.8.0'

## Expected Behavior
To have no error when building with grunt build
Is there a way to make it run on my rpi? I know its stated its not supported but maybe there is a way? 

## Current Behavior
Page with no elements except some button and text element is shown

## Context
Running pokegomap on my raspberry pi.

## Your Environment
* Version used: Latest dev branch (commit: a83ce5c0e6bfa4a97df485498e27fe3a0181ef7a)
* Environment name and version (e.g. Python 2.7): Python 2.7
* Operating System and version (desktop or mobile):Raspian Jessy

**Comment by Imoq** [comment]
Subscribing as I have a related issue on Raspberry Pi, trying to run devel (main branch runs ok):

> root@raspberrypi:~/PoGo#  python runserver.py -st 5 -sd 5 -t 1 -dc
2016-08-01 00:23:39,195 [    MainThread][     runserver] [CRITICAL] Please run "grunt build" before starting the server.
root@raspberrypi:~/PoGo# grunt build
Traceback (most recent call last):
  File "/usr/local/bin/grunt", line 9, in <module>
    load_entry_point('grunt==0.1.1', 'console_scripts', 'grunt')()
  File "/usr/local/lib/python2.7/dist-packages/grunt/render.py", line 94, in main
    render_from_file(**options)
  File "/usr/local/lib/python2.7/dist-packages/grunt/render.py", line 54, in render_from_file
    with codecs.open(input, encoding=encoding) as f:
  File "/usr/lib/python2.7/codecs.py", line 881, in open
    file = __builtin__.open(filename, mode, buffering)
IOError: [Errno 2] No such file or directory: 'build'
root@raspberrypi:~/PoGo# uname -a
Linux raspberrypi 3.18.11-v7+ #781 SMP PREEMPT Tue Apr 21 18:07:59 BST 2015 armv7l GNU/Linux
root@raspberrypi:~/PoGo#

**Comment by turtle2472** [comment]
I'm really sorry that I'm not going to be much help on this specific issue.  I did get the maps running on my RPi but I don't know exactly how.  

It would be nice to have an actual install guide for an RPi since it seems to be pretty specific compared to other OS installs.

**Comment by WtfJoke** [comment]
Some googling did the job.
After running 
'npm rebuild node-sass'
and another
'grunt build'
everything works as expected :)

**Comment by turtle2472** [comment]
Mind specifying your install process?  I'm thinking about using another RPi I have and making it just run maps.  While I could figure it out myself, it would be nice to have it documented somewhere.

**Comment by WtfJoke** [comment]
I will post it tomorrow if I can recall most of the things. Now I need some sleep :sleepy:
I think my install process didnt differ much from the part of the wiki except the npm/grunt stuff.

**Comment by coolaj86** [comment]
FYI: I've got a node.js fork that will run an an RPi without a hitch https://github.com/Daplie/node-pokemap

It doesn't use sass or python C-modules or any additional build tools or build steps - it's all plain JavaScript.

**Comment by WtfJoke** [comment]
My install process (as far as I remember):
## Before you start
```
sudo apt-get update
sudo apt-get dist-upgrade
```

## General

```
git clone https://github.com/AHAAAAAAA/PokemonGo-Map.git
git checkout develop
cd PokemonGo-Map/config/
cp config.ini.example  config.ini
nano config.ini
adding gmaps key + save and exit
```

## Prerequirements

### Getting nodejs (optional?)
```
curl -sL https://deb.nodesource.com/setup_6.x | sudo -E bash -
sudo apt-get install -y nodejs
```

### PogoWikiPart
```
sudo apt-get install python python-pip nodejs npm nodejs-legacy
cd PokemonGo-Map/
pip install -r requirements.txt
pip install -r requirements.txt --upgrade
npm install
```

### Grunt
```
npm install grunt
npm install
```

Maybe optional: 
```
npm install -g grunt-cli
npm install
```

### Make server runnable
```
npm rebuild node-sass
grunt build

```

finally you should be able to execute runserver.py without problems.

If you encounter any problems, let me know.

**Comment by Imoq** [comment]
Thanks for the detailed instructions!

Unfortunately, it's still failing for me. The complete (large!) output is here: http://pastebin.com/EY507TSh

But in summary, what fails (no matter what I do) is always:

> make: Leaving directory '/root/PoGoNew/node_modules/grunt-sass/node_modules/node-sass/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/root/PoGoNew/node_modules/grunt-sass/node_modules/node-sass/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Linux 3.18.11-v7+
gyp ERR! command "/usr/local/bin/node" "/root/PoGoNew/node_modules/grunt-sass/node_modules/node-sass/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--verbose" "--libsass_ext=" "--libsass_cflags=" "--libsass_ldflags=" "--libsass_library="
gyp ERR! cwd /root/PoGoNew/node_modules/grunt-sass/node_modules/node-sass
gyp ERR! node -v v4.4.7
gyp ERR! node-gyp -v v3.4.0
gyp ERR! not ok 
Build failed
npm ERR! Linux 3.18.11-v7+
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install"
npm ERR! node v4.4.7
npm ERR! npm  v2.15.8
npm ERR! code ELIFECYCLE

>npm ERR! node-sass@3.8.0 postinstall: `node scripts/build.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the node-sass@3.8.0 postinstall script 'node scripts/build.js'.
npm ERR! This is most likely a problem with the node-sass package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node scripts/build.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs node-sass
npm ERR! Or if that isn't available, you can get their info via:
npm ERR! 
npm ERR!     npm owner ls node-sass
npm ERR! There is likely additional logging output above.

>npm ERR! Please include the following file with any support request:
npm ERR!     /root/PoGoNew/npm-debug.log

**Comment by WtfJoke** [comment]
Whats your output on node --version? 
From log looks like you have 4.4.7, where I have 6.3.1 installed (thats the only difference I see atm.

**Comment by Imoq** [comment]
Yup! 4.4.7., I'll see if there is a legacy (?) node install that is causing some issues and re-try.

root@raspberrypi:~/new/PokemonGo-Map# node --version
v4.4.7
root@raspberrypi:~/new/PokemonGo-Map# 

Thanks!

**Comment by Imoq** [comment]
I believe the issue I have is that npm I get from that repository is only 1.1.4:

>root@raspberrypi:~/new/PokemonGo-Map# npm -v
1.1.4
root@raspberrypi:~/new/PokemonGo-Map# 

>npm ERR! System Linux 3.18.11-v7+
npm ERR! command "/usr/bin/nodejs" "/usr/bin/npm" "install"
npm ERR! cwd /root/new/PokemonGo-Map
npm ERR! node -v v0.6.19
npm ERR! npm -v 1.1.4
npm ERR! message failed to fetch from registry: grunt-contrib-clean

I must be overlooking something, I cleansed everything from /usr/local/ from the tarball installation I had attempted last night.

**Comment by WtfJoke** [comment]
I have npm version 3.10.3 installed

**Comment by Imoq** [comment]
I need to delete all traces of any conflicting node/npm versions I may have, as I attempted various installations last night and nothing seemed to work. I will be tempted to even wipe out and start from scratch on the Raspberry Pi since I don't use it for anything (it was collecting dust before the weekend).

I guess I'll try a bit more and see if I can figure out how to delete any trace of any stale node/npm before giving up :)

Thanks!

**Comment by Imoq** [comment]
What repository did you install npm from?

I have this:

>root@raspberrypi:~/new/PokemonGo-Map# apt-cache policy npm
npm:
  Installed: 1.1.4~dfsg-2
  Candidate: 1.1.4~dfsg-2
  Version table:
 *** 1.1.4~dfsg-2 0
        500 http://mirrordirector.raspbian.org/raspbian/ wheezy/main armhf Packages
        100 /var/lib/dpkg/status
root@raspberrypi:~/new/PokemonGo-Map#

**Comment by WtfJoke** [comment]
> start from scratch on the Raspberry Pi since I don't use it for anything (it was collecting dust before the weekend).

I actually did the same yesterday^^ 


Thats my output
```
pi@raspberrypi:~/git/PokemonGo-Map $ apt-cache policy npm
npm:
  Installed: (none)
  Candidate: 1.4.21+ds-2
  Version table:
     1.4.21+ds-2 0
        500 http://mirrordirector.raspbian.org/raspbian/ jessie/main armhf Packages
        100 /var/lib/dpkg/status
```

**Comment by Imoq** [comment]
So, interesting, you are in jessie and I am in wheezy. I'll look further into this. Thanks again! :)

**Comment by WtfJoke** [comment]
Yeah, I wanted to install wheezy, but didnt found it on the official page anymore. So I thought I give it a try.

**Comment by Imoq** [comment]
Went out to buy a micro SSD card just to install Jessie and not losing my current (working) main branch radar. A friend recommended me to use noobs, so I am doing it. Will report back soon whenever I have it (hopefully) up and running, to confirm your installation instructions :)

**Comment by Imoq** [comment]
ok, got it up and running (finally!) but I had to do some weird steps and specially to repeat few things in order for this to be working on Raspberry Pi. Getting most of the commands and order from bash history.

1. Installed a brand new jessie on the SSD.
2. apt-get update
3. apt-get upgrade
4. git clone -b develop https://github.com/AHAAAAAAA/PokemonGo-Map.git
5. curl -sL https://deb.nodesource.com/setup_6.x | bash -
6. apt install git nodejs npm python-pip nodejs-legacy python-dev  (apt complained about nodejs-legacy conflicting to already installed version, so I had to remove it from the command and re-run).
7. pip install -r requirements.txt <--- installed few of them, but had error message at the end. Retried a second time, still an error. Decided to install manually each of the ones described on the .txt, and it worked!, so:
8. pip install six
9. pip install flask
10. pip install jinja2
11. pip install markupsafe
12. pip install configargparse
13. pip install click
14. pip install itsdangerous
15. pip install peewee
16. pip install wsgiref
17. pip install geopy
18. pip install protobuf
19. pip install requests
20. pip install s2sphere
21. pip install gpsoauth
22. pip install protobuf-to-dict
23. pip install pymysql
24. pip install flask-cors
25. pip install flask-compress
26. pip install latlon
27. After the last one, just re-ran: pip install -r requirements.txt and it worked and installed whatever was missed.
28. npm install
29. npm install -g grunt-cli
30. npm install node-sass
31. grunt build

Success! The process is **very long** (specially on the apt-get update/upgrade and on the compiling of many of the stuff) but after few hours, I have the devel branch running on the Raspberry Pi.

Thanks a lot for the help! :)

**Comment by WtfJoke** [comment]
Cool, great you made it. Should we place this somewhere in the wiki?
